apps/eew/sceewlog/eews2fcm.py

```diff
@@ -36,31 +35,71 @@ def __init__(self, fcmDataFile, worldCitiesFile, language = "es-US"):
         self.language = language
         self.agency = ''
         self.distance = 0
-    
+        self.projectId = "" 
+        self.serviceAccountFile = ""
+
     def readFcmDataFile(self):
         if os.path.exists( self.fcmDataFile ):
             config = configparser.RawConfigParser()
-            
+
         else:
             seiscomp.logging.error("the file %s does not exists" % self.fcmDataFile )
             return -1
-        
+
         try:
             config.read(self.fcmDataFile)
         except Exception as e:
             seiscomp.logging.error("Not possible to open %s " % self.fcmDataFile)
             return -1
-            
+
         try:
             self.authKey = config.get('AUTHKEY', 'key')
             self.topic = config.get('TOPICS', 'topic')
+            self.serviceAccountFile = config.get('SERVICEFILE', 'servicefile')
+            self.projectId = config.get('PROJECTID', 'projectid')
         except:
             seiscomp.logging.error('Not possible to parse the configuration file.')
             return -1
-    
+    def get_access_token(self):
+        """Retrieve a valid access token that can be used to authorize requests.
+
+        :return: Access token.
+        """
+        SCOPES=["https://www.googleapis.com/auth/firebase.messaging"] # only for notifications
+        credentials = service_account.Credentials.from_service_account_file(
+          self.serviceAccountFile, scopes=SCOPES) 
```

Comment:
Actually a token is obtained using the service JSON file and this is not stored anywhere because this is for using once although it lasts for a hour. So, we only provide the service JSON file in the configuration.

Reply:
OK closing this discussion, and following it up on the next one :)